### PR TITLE
Add healthz endpoint

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#1902

Adds the healthz kubernetes endpoint and gets some of the client-go deps sorted. Sorry its a bit big due to vendor.

screen shot 2018-02-28 at 12 44 45 pm
 LOL now it's becoming a real operator :trollface:

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -45,6 +57,14 @@
   name = "github.com/spf13/viper"
   version = "1.0.0"
 
-[prune]
-  go-tests = true
-  unused-packages = true
```
[the remark this answers is not in the document]
Comment (fgimenez) [reply Gopkg.toml]: AFAIK this is a default setting with the latest dep version, these are the docs https://golang.github.io/dep/docs/Gopkg.toml.html#prune, it was added recently to e2e harness and reduced the number of files in vendor a lot https://github.com/giantswarm/e2e-harness/pull/19/files. I've given it a try in this branch and it is pretty amazing:
```
$ git ls-files --deleted | wc -l
4324
```

[the remark this answers is not in the document]
Comment (indyfree) [reply Gopkg.toml]: Pruning seems like a sane thing to do. I like it

**Comment by rossf7** [comment]
I reenabled prune and it does makes a big difference.

screen shot 2018-02-28 at 2 12 25 pm

Bit off topic but my favourite delete is this one. An 11MB PDF from 10 years ago! ;)

https://github.com/petar/GoLLRB/blob/master/doc/Sedgewick-Talk-Penn2008.pdf

File: Gopkg.toml [hunk 2]
```diff
@@ -24,6 +24,14 @@
 #   go-tests = true
 #   unused-packages = true
 
+required = [
+  "k8s.io/api/core/v1",
+  "k8s.io/apimachinery/pkg/apis/meta/v1",
+]
```
[the remark this answers is not in the document]
Comment (rossf7) [reply Gopkg.toml]: The error usually indicates something is wrong with the `k8s.io/client-go` dependencies. We're pinning to the `kubernetes-1.8.5` branch which is what we're using in `operatorkit` at the moment.

See https://gigantic.slack.com/archives/C0251MXL5/p1516957715000216 for more details.

[the remark this answers is not in the document]
Comment (indyfree) [reply Gopkg.toml]: Thanks!

**Comment by rossf7** [comment]
Going with `prune`. The size difference is substantial and it's setup by `dep init`. IMO its worth using in all operators.

**Pull request merged**